### PR TITLE
Add a test to verify nonzero correctness.

### PR DESCRIPTION
**Pull request by vanbasten23**
Add a test to catch issues such as https://github.com/pytorch/xla/issues/4432.

### CODE REVIEW

File: test/ds/test_dynamic_shapes.py [hunk 1]
```diff
@@ -168,6 +168,13 @@ def test_nonzero_shape(self):
         torch.nonzero(x, as_tuple=False), 0)
     self.assertEqual(x_dim0_shape.item(), 4)
 
+  def test_nonzero_correctness(self):
+    t1 = torch.tensor([[1, 0, 0, 5, 0, 6]], device=dev)
+    t2 = torch.nonzero(t1)
+    t1_aten = torch.tensor([[1, 0, 0, 5, 0, 6]])
```
Comment (JackCaoG) [inline test/ds/test_dynamic_shapes.py:174]: `t1_aten =t1.cpu()` maybe?

File: test/ds/test_dynamic_shapes.py [hunk 2]
```diff
@@ -168,6 +168,13 @@ def test_nonzero_shape(self):
...
+    t2_aten = torch.nonzero(t1_aten)
+    self.assertEqual(t2.cpu(), t2_aten.cpu())
```
Comment (JackCaoG) [inline test/ds/test_dynamic_shapes.py:176]: `t2_aten` is already on cpu

File: test/ds/test_dynamic_shapes.py (hunk 1 above)
Comment (vanbasten23) [reply test/ds/test_dynamic_shapes.py:174]: done

File: test/ds/test_dynamic_shapes.py (hunk 2 above)
Comment (vanbasten23) [reply test/ds/test_dynamic_shapes.py:176]: done

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**